akka-docs/src/main/paradox/typed/persistence-style.md

```diff
@@ -0,0 +1,51 @@
+# Persistence - coding style
+
+## Event handlers in the state
+
+The section about @ref:[Changing Behavior](persistence.md#changing-behavior) described how commands and events
+can be handled differently depending on the state. One can take that one step further and define the event
+handler inside the state classes. In @ref:[next section the command handlers](#command-handlers-in-the-state) are
+also defined in the state.
+
+The state can be seen as your domain object and it should contain the core business logic. Then it's a matter
+of taste if event handlers and command handlers should be defined in the state or be kept outside it.
+
+Here we are using a bank account as the example domain. It has 3 state classes that are representing the lifecycle
+of the account; `EmptyAccount`, `OpenedAccount`, and `ClosedAccount`.
+
+Scala
+:  @@snip [AccountExampleWithEventHandlersInState.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExampleWithEventHandlersInState.scala) { #account-entity }
+
+TODO include corresponding example in Java
```

Comment:
Issue before merging?

Reply:
Let's do that in a separate PR. I didn't want to touch Java now, because Renato is working on new (mutable) builders.

Reply:
:+1: